cardano-explorer/src/Explorer/Web/ClientTypes.hs

```diff
@@ -0,0 +1,264 @@
+
+--instance NFData CBlockEntry
+
+-- | List of tx entries is returned from "get latest N transactions" endpoint
+data CTxEntry = CTxEntry
+    { cteId         :: !CTxId
+    , cteTimeIssued :: !(Maybe POSIXTime)
+    , cteAmount     :: !CCoin
+    } deriving (Show, Generic)
+
+
+data CBlockRange = CBlockRange
+    { cbrBlocks     :: [CBlockSummary]
+    , cbrTransactions :: [CTxSummary]
+    } deriving (Show, Generic)
+
+-- | Data displayed on block summary page
+data CBlockSummary = CBlockSummary
+    { cbsEntry      :: !CBlockEntry
+    , cbsPrevHash   :: !CHash
+    , cbsNextHash   :: !(Maybe CHash)
+    , cbsMerkleRoot :: !CHash
+    } deriving (Show, Generic)
+
+data CAddressType
+    = CPubKeyAddress
+    | CScriptAddress
+    | CRedeemAddress
+    | CUnknownAddress
+    deriving (Show, Generic)
+
+data CAddressSummary = CAddressSummary
+    { caAddress :: !CAddress
+    , caType    :: !CAddressType
+    , caTxNum   :: !Word
+    , caBalance :: !CCoin
+    , caTxList  :: ![CTxBrief]
+    } deriving (Show, Generic)
+
+data CTxBrief = CTxBrief
+    { ctbId         :: !CTxId
+    , ctbTimeIssued :: !(Maybe POSIXTime)
+    , ctbInputs     :: ![Maybe (CAddress, CCoin)]
+    , ctbOutputs    :: ![(CAddress, CCoin)]
+    , ctbInputSum   :: !CCoin
+    , ctbOutputSum  :: !CCoin
+    } deriving (Show, Generic)
+
+data CUtxo = CUtxo
+    { cuId       :: !CTxId
+    , cuOutIndex :: !Int
+    , cuAddress  :: !CAddress
+    , cuCoins    :: !CCoin
+    }
+    | CUtxoUnknown
+    { cuTag  :: !Int
+      , cuBs :: !CByteString
+    }
+    deriving (Show, Generic)
+
+newtype CNetworkAddress = CNetworkAddress Text
+    deriving (Show, Generic)
+
+data CTxSummary = CTxSummary
+    { ctsId              :: !CTxId
+    , ctsTxTimeIssued    :: !(Maybe POSIXTime)
+    , ctsBlockTimeIssued :: !(Maybe POSIXTime)
+    , ctsBlockHeight     :: !(Maybe Word)
+    , ctsBlockEpoch      :: !(Maybe Word64)
+    , ctsBlockSlot       :: !(Maybe Word16)
+    , ctsBlockHash       :: !(Maybe CHash)
+    , ctsRelayedBy       :: !(Maybe CNetworkAddress)
+    , ctsTotalInput      :: !CCoin
+    , ctsTotalOutput     :: !CCoin
+    , ctsFees            :: !CCoin
+    , ctsInputs          :: ![Maybe (CAddress, CCoin)]
+    , ctsOutputs         :: ![(CAddress, CCoin)]
+    } deriving (Show, Generic)
+
+data CGenesisSummary = CGenesisSummary
+    { cgsNumTotal               :: !Int
+    , cgsNumRedeemed            :: !Int
+    , cgsNumNotRedeemed         :: !Int
+    , cgsRedeemedAmountTotal    :: !CCoin
+    , cgsNonRedeemedAmountTotal :: !CCoin
+    } deriving (Show, Generic)
+
+data CGenesisAddressInfo = CGenesisAddressInfo
+    { cgaiCardanoAddress :: !CAddress
+    , cgaiGenesisAmount  :: !CCoin
+    , cgaiIsRedeemed     :: !Bool
+    } deriving (Show, Generic)
+
+data CAddressesFilter
+    = RedeemedAddresses
+    | NonRedeemedAddresses
+    | AllAddresses
+    deriving (Show, Generic)
+
+instance Default CAddressesFilter where
+    def = AllAddresses
```

Comment:
I think the `Default` type class is a really bad idea (explicit  is better than implicit). Unless you know exactly why you are using it, we should probably drop it.

Reply:
it was used by `DQueryParam` which provides a default if the user doesnt supply one in the url
porting it over was tricky, so i just changed all of those to `Maybe`'s